core/chains/evm/logpoller/log_poller.go

```diff
@@ -659,12 +659,11 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 		gethLogs, err := lp.ec.FilterLogs(ctx, lp.Filter(big.NewInt(from), big.NewInt(to), nil))
 		if err != nil {
 			var rpcErr client.JsonError
-			if errors.As(err, &rpcErr) {
-				if rpcErr.Code != jsonRpcLimitExceeded {
-					lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
-					return err
-				}
+			if !errors.As(err, &rpcErr) || rpcErr.Code != jsonRpcLimitExceeded {
+				lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
```

Comment:
`if !errors.As(err, &rpcErr) || ...` is true when `err == nil`.

Should it be this?
```suggestion
			if errors.As(err, &rpcErr) && rpcErr.Code != jsonRpcLimitExceeded {
				lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
```

Reply:
No, that's actually how it is currently is on develop branch, and that's the bug that's being fixed here:

Presently, it checks to see if it's an rpc error AND if it's not jsonRpcLimitExceeded.  But it should be checking that it's either not an rpc error or not the specific rpc error jsonRpcLimitExceeded.  In either of those cases, it should just return the error.  The only case where it needs to handle it specially (by reducing the batch size is if it's specifically this type of rpc error.

The nil case is already handled, because this is in an `if err != nil` block

I guess it might be more readable if instead of `!errors.As() || rpc.Err.Code != jsonRpcLimitExceeded` we change it to `!(errors.As() && rpc.Err.Code == jsonRpcLimitExceeded)`?

Reply:
If the `As` check fails, then you cannot access the `rpcErr.Code` field. This does not make sense as written.

Reply:
Oh, I think I see what this is doing but it seems like a messy way to express it.

Reply:
What about this?
```suggestion
			if !(errors.As(err, &rpcErr) && rpcErr.Code == jsonRpcLimitExceeded) {
				lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
```

Reply:
Yeah, that's what I suggested above... I think that's a better way of expressing it.

Reply:
This logic got slightly more complex once I added a second error code it can match against, so I moved it into a separate helper function where I think it's a lot easier to read.